gensim/models/ldamodel.py

```diff
@@ -900,10 +900,26 @@ def get_document_topics(self, bow, minimum_probability=None):
         if is_corpus:
             return self._apply(corpus)
 
-        gamma, _ = self.inference([bow])
+        gamma, phis = self.inference([bow], collect_sstats=True)
         topic_dist = gamma[0] / sum(gamma[0])  # normalize distribution
-        return [(topicid, topicvalue) for topicid, topicvalue in enumerate(topic_dist)
-                if topicvalue >= minimum_probability]
+
+        if per_word_topics is False:
+            return [(topicid, topicvalue) for topicid, topicvalue in enumerate(topic_dist)
+                    if topicvalue >= minimum_probability]
+
+        elif per_word_topics is True:
+            word_phi = [] # contains word and corresponding topic
+            for word_type, weight in bow:
+                phi_values = [] # contains phi values for each topic
+                for topic_id in range(0, self.num_topics):
+                    if phis[topic_id][word_type] >= minimum_probability:
+                        phi_values.append((topic_id, phis[topic_id][word_type]))
+                    # appends phi values for each topic for that word
+                word_phi.append((word_type, phi_values))
+                # appends the word and the corresponding values for each topic.
+                # If we choose the highest value for a word, we get the corresponding most probable topic.
+            return ([(topicid, topicvalue) for topicid, topicvalue in enumerate(topic_dist)
+                    if topicvalue >= minimum_probability], word_phi) # returns 2-tuple
```

Comment:
What I think is not terribly important... it'd be more valuable to ask the users who asked for this feature.

When I needed this functionality, I needed to highlight words from different topics in different colours, for visualization. So an API that facilitates that goal would have helped; a word_id=>topic_id mapping for a particular document would have been helpful.

So that's one data point :)